### ISSUE TITLE
User Creation: Allow user, contact & place in one step for multiple users

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-core#7503]
**What feature do you want to improve?**
Using the [UI today](https://docs.communityhealthtoolkit.org/apps/tutorials/contact-and-users-1/#implementation-steps), to add a CHW, you first have to create their place and their contact in the CHT app, and then go over Admin app and create their user, selecting the contact and place you just created.  There is the additional overhead of delivering the credentials after the account has been created. 

**Describe the improvement you'd like**
It would be great if admins could have a way to do this in fewer steps to streamline the account creation. To make this really amazing, if the CHT could [automatically send a magic link](https://docs.communityhealthtoolkit.org/apps/concepts/access/#magic-links-for-logging-in) to the user.

Some features we should accommodate in the UX:
* error handling in tabular input (missing fields, unique phone numbers)
* allowing use to skip rows (fill out 2nd and 4th rows only) and still save data
* explanatory labels in admin form should be configurable, thus taken from translations

**Describe alternatives you've considered**
Use the existing manual process.

**Additional context**
A number of partners have expressed interested in having the ability of the respective Ministries of Health being able to do this with out the use of a command line in an efficient manner for a small number of users at a time. 

Per the continuous discover process: 
* [story map](https://docs.google.com/presentation/d/1XufQZEShpWXPel9azisVQlPPaoPpufpKijkRMhAw8K8/edit#slide=id.gfbfc44f19a_0_0) 
* [solution selection](https://docs.google.com/document/d/18MN-e2z-OwcHBE2EvfH5yIKHnVCL7hWvvncYtNeF81I/edit#heading=h.g1sek8njjyb7)
* [Interviews](https://drive.google.com/drive/folders/1x0ef7bXn2wytCCLFAmFw3tKovx2ZPoEi)

**Why Now?**
Partner(s): Guava Pineapple
Reason(s): A partner will be blocked by this for a major release or expansion

**Comment by mrjones-plip** [comment]
Assigning to @n-orlowski to do UI/UX work on this and putting it into "Dev In progress".  After UI/UX work is done and approved, we'll move it over to @m5r  for coding.

**Comment by mrjones-plip** [comment]
@abbyad  @michaelkohn @n-orlowski - added "Some features we should accommodate in the UX" in ticket body - feel free to add to this while we're UX prototype mode!

**Comment by mrjones-plip** [comment]
@m5r - while the design is still a bit still in-flight, I think it's important to make sure you're not blocked by the earlier development (any missing APIs etc) for this feature.  Assigning this to you too then!

Please let me know if you're blocked by anything - I want to clear the path for you!

**Comment by garethbowen** [comment]
Not a blocker for 3.15.0

**Comment by latin-panda** [comment]
API work:

- [ ] Add CSV support to [/api/v1/users](https://docs.communityhealthtoolkit.org/apps/reference/api/#post-apiv1users)
- [ ] Parse CSV to JSON
- [ ] Stream line log progress in `medic-log`, doc name might be: `bulk-user-upload-[yyyymmddHHMMss]`
- [ ] Maybe make way to export this log to CSV from API... otherwise from admin app 🤔

**Comment by mrjones-plip** [comment]
@latin-panda  - I'm working with @tatilepizs and we can't get this to work as of [5863cc3](https://github.com/medic/cht-core/pull/7614/commits/5863cc391ac723dafa8ed4f1602b59c38b2b9c04). Here's the steps we took to reproduce

1. check out `7503-bulk-user-creationg-api` branch and run `grunt` in main folder and `node server.js` once for `api` dir and once for `sentinel` dir
2. create a test CSV file called `7503.sample.csv` with this contents based on [your test lines](https://github.com/medic/cht-core/pull/7614/files#diff-29c8f8b69bb63bf0956dc153117fc26850a9c1b5118ae6cddd333d3ab6f763aeR2551):
   ```
   password,sername,type,place,contact.name,contact.phone,contact.address
   Secret1234,mary,person,498a394e-f98b-4e48-8c50-f12aeb018fcc,mary,2652527222,"1 King ST, Kent Town, 55555"
   Secret5678, peter ,person,498a394e-f98b-4e48-8c50-f12aeb018fcc,Peter, 2652279,"15 King ST, Kent Town, 55555 "
   ```
3. `POST` it with `curl` like so and get back an empty JSON response (`[]`):
   ```
   curl  -H "Content-Type: text/csv" --request POST -d @7503.sample.csv  http://medic:password@localhost:5988/api/v1/users
   []
   ```
   
Here's a video in case it's helpful:

https://user-images.githubusercontent.com/8253488/166338332-84b32f5b-ada4-41a8-836c-cc44f374d5b0.mp4

I'll try some JSON next instead of CSV maybe?

**Comment by mrjones-plip** [comment]
Aha! I got it to work after some JSON hacking. I get nice contacts with the correct parents and I get users with the correct contact and place selected.  So this call:

```
curl  -H "Content-Type: application/json" --request POST -d @deleteme.json  http://medic:password@localhost:5988/api/v1/users
```

works with this JSON:

```json
[

  {
    "password": "Medic3212",
    "type": "district-manager",
    "username": "lisa4",
    "place": "262e09d8-f34e-473f-9c00-2775465f4742",
    "contact": {
      "name": "lisa Smith4",
      "phone": "+2868917046",
      "place": "262e09d8-f34e-473f-9c00-2775465f4742"
    }
  },
  {
    "password": "Medic3212",
    "username": "tom4",
    "type": "district-manager",
    "place": "262e09d8-f34e-473f-9c00-2775465f4742",
    "contact": {
      "name": "Tom Johnson4",
      "phone": "+2868917046",
      "place": "262e09d8-f34e-473f-9c00-2775465f4742"
    }
  }
]
```

The API endpoint only accepts JSON and the admin UI will convert the CSV to JSON for the user before calling the API I assume?

Separately, I found an error if I have an extra `,` in there on the 2nd to last line in the JSON above like this I get back a `500` and an error is logged. While not a problem to return a `500` for this MVP it _is_ a problem that we're logging passwords in the clear:

```
REQ e34af708-685d-4a5a-ad64-8ca969b72014 ::ffff:127.0.0.1 medic POST /api/v1/users HTTP/1.1
2022-05-02 15:57:44 ERROR: Server error: SyntaxError: Unexpected token ] in JSON at position 547
    at JSON.parse (<anonymous>)
    at parse (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/body-parser/lib/types/json.js:89:19)
    at /home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/body-parser/lib/read.js:121:18
    at invokeCallback (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/raw-body/index.js:224:16)
    at done (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/raw-body/index.js:213:7)
    at IncomingMessage.onEnd (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/raw-body/index.js:273:7)
    at IncomingMessage.emit (events.js:228:7)
    at endReadableNT (_stream_readable.js:1185:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  [stack]: 'SyntaxError: Unexpected token ] in JSON at position 547\n' +
    '    at JSON.parse (<anonymous>)\n' +
    '    at parse (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/body-parser/lib/types/json.js:89:19)\n' +
    '    at /home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/body-parser/lib/read.js:121:18\n' +
    '    at invokeCallback (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/raw-body/index.js:224:16)\n' +
    '    at done (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/raw-body/index.js:213:7)\n' +
    '    at IncomingMessage.onEnd (/home/mrjones/Documents/MedicMobile/cht-core/api/node_modules/raw-body/index.js:273:7)\n' +
    '    at IncomingMessage.emit (events.js:228:7)\n' +
    '    at endReadableNT (_stream_readable.js:1185:12)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:81:21)',
  [message]: 'Unexpected token ] in JSON at position 547',
  expose: true,
  statusCode: 400,
  status: 400,
  body: '[  {    "password": "Medic3212",    "type": "district-manager",    "username": "lisa4",    "place": "262e09d8-f34e-473f-9c00-2775465f4742",    "contact": {      "name": "lisa Smith4",      "phone": "+2868917046",      "place": "262e09d8-f34e-473f-9c00-2775465f4742"    }  },  {    "password": "Medic3212",    "username": "tom4",    "type": "district-manager",    "place": "262e09d8-f34e-473f-9c00-2775465f4742",    "contact": {      "name": "Tom Johnson4",      "phone": "+2868917046",      "place": "262e09d8-f34e-473f-9c00-2775465f4742"    }  },]',
  type: 'entity.parse.failed'
} 
RES e34af708-685d-4a5a-ad64-8ca969b72014 ::ffff:127.0.0.1 medic POST /api/v1/users HTTP/1.1 500 - 4.679 ms
```

cc @tatilepizs  - it'd be good for you to be able to reproduce these results and do any further testing you see fit!  I'm also available to help do more QA assist tomorrow afternoon as needed.

**Comment by mrjones-plip** [comment]
I think I found a bug? Maybe two!

1. If a place has no contacts and we're adding one, it should be the default contact.  If it has contacts, and we're adding new ones, we shouldn't make the  n+1  contact the default contact.
2. After installing [config-moh-mali-cht config](https://github.com/medic/config-moh-mali-chw) and `upload-docs` from [step 3 and 4 with the test JSON](https://docs.google.com/document/d/1FKowuecQNwxOZrx1ktDgWRhl3yJnCMicNZHOt3ot9So/edit#heading=h.nimljpt9tx6o), if I run this JSON against the API, I only see one contact created in the place with the ID of `955db154-244b-520e-bb31-b7053e7ef1ef`. Further, clicking the contact in the UI results in error `No person selected`:
   ```json
    [
      {
        "password": "Medic3212",
        "type": "chw",
        "username": "sally1",
        "phone": "+2868917046",
        "place": "e0891f25-1a3f-58f0-92f3-ab0f33bc75ca",
        "contact": {
          "name": "sally Smith",
          "phone": "+2868917046",
          "type": "c62_chw",
          "place": "e0891f25-1a3f-58f0-92f3-ab0f33bc75ca"
        }
      },
      {
        "password": "Medic3212",
        "username": "thomas1",
        "phone": "+2868917046",
        "type": "chw",
        "place": "e0891f25-1a3f-58f0-92f3-ab0f33bc75ca",
        "contact": {
          "name": "thomas Johnson",
          "phone": "+2868917046",
          "type": "c62_chw",
          "place": "e0891f25-1a3f-58f0-92f3-ab0f33bc75ca"
        }
      }
    ]

   ```
   
   Here's the exact call and response for easy debugging. It suggests it inserted valid contacts and users:
   
   ```
   curl  -H "Content-Type: application/json" --request POST -d @deleteme.json  http://medic:password@localhost:5988/api/v1/users
   [{"contact":{"id":"f661b9ae8a3b69e50dc658981500ae7a","rev":"1-65706fbb75324a2bd7431ace053224f5"},"user":{"id":"org.couchdb.user:sally1","rev":"1-142e8b563e3e68c308269aa42afb0353"},"user-settings":{"id":"org.couchdb.user:sally1","rev":"1-a20edf416f83efd2742ed8dd86b5014f"}},{"contact":{"id":"f661b9ae8a3b69e50dc658981500a68c","rev":"1-808b8a0c0a94e4a68a6148680d5059ef"},"user":{"id":"org.couchdb.user:thomas1","rev":"1-6ad536d2d6963289ab39fca43339b3e8"},"user-settings":{"id":"org.couchdb.user:thomas1","rev":"1-085ef08697274554beda7d0fe1a5031a"}}]%    
   ```

**Comment by latin-panda** [comment]
Thanks @mrjones-plip, this is more related to the pre-existing implementation of the multi-user upload, I'm going to read the code and find what's going on. The json or csv should be very well formatted, for example the place should exists, fields names should match and be in the same order as the values, however I'm not sure how the default contact is working in this API, I'll find out and let you know.

**Comment by latin-panda** [comment]
For anyone testing, it'd be very helpful to also paste here the API response please 🙏, if there's an error in the structure then it should (in theory) return an error.

**Comment by latin-panda** [comment]
[#1 this comment ](https://github.com/medic/cht-core/issues/7503#issuecomment-1115441054)

@mrjones-plip, I created this document in [Google Sheets](https://github.com/medic/cht-core/issues/7503#issuecomment-1115441054), downloaded it as csv (see attachment [test-user-creation - Sheet1 (3).csv](https://github.com/medic/cht-core/files/8608787/test-user-creation.-.Sheet1.3.csv)) then I used postman to select the file and upload it, and got successful response. 
 
<img width="500" alt="Screen Shot 2022-05-03 at 10 02 02 am" src="https://user-images.githubusercontent.com/66472237/166415816-13e6a87d-c1e7-4918-a3e0-047b524ffefc.png">

Maybe you can try this one that I know it's working, just to discard something like special character in Windows making errors (since I'm testing in Mac), please change the place id with one existing in your instance.  Thanks

**Comment by latin-panda** [comment]
[#2 this comment](https://github.com/medic/cht-core/issues/7503#issuecomment-1115448098) about passwords in logs.

It's not nice but not critical since the record wasn't saved, I mean the password doesn't exist in the DB yet and it's not from an active user. The bodyparser is throwing this error before it reaches our controller code, it's happening for other endpoints in master as well. 
@garethbowen is this something we want to fix in master? I can create a ticket and get it prioritised.  <- I suggest this.

**Comment by latin-panda** [comment]
[#3 this comment](https://github.com/medic/cht-core/issues/7503#issuecomment-1115755887) about MOH Mali config.

> If a place has no contacts and we're adding one, it should be the default contact. If it has contacts, and we're adding new ones, we shouldn't make the n+1 contact the default contact.

I've test it in 3.14.x and 3.15.x, using single and multiple user upload. Every time the new user is created using `api/v1/users` it changes the primary contact of the place to this newly created user. It's an old behaviour that probably need to be reconsidered, but as for now -> it's out of scope for this FR.

> After installing [config-moh-mali-cht config](https://github.com/medic/config-moh-mali-chw) and upload-docs from [step 3 and 4 with the test JSON](https://docs.google.com/document/d/1FKowuecQNwxOZrx1ktDgWRhl3yJnCMicNZHOt3ot9So/edit#heading=h.nimljpt9tx6o), if I run this JSON against the API, I only see one contact created in the place with the ID of 955db154-244b-520e-bb31-b7053e7ef1ef. Further, clicking the contact in the UI results in error No person selected

and 

> When I look in the admin area the thomas1 user doesn't have thomas Johnson as it's associated contact. It does have correct username, phone, role, and place though.

It was creating all the required docs in the DB, but just not with the right fields. So cht-core was having some issues with hydrating the user and getting the contact's lineage:

<img width="800" alt="Screen Shot 2022-05-03 at 12 59 16 pm" src="https://user-images.githubusercontent.com/66472237/166456243-9484ef32-dff7-469a-8f43-08aee5c0ebe8.png">
<img width="800" alt="Screen Shot 2022-05-03 at 12 58 54 pm" src="https://user-images.githubusercontent.com/66472237/166456261-77cfca3b-e773-441d-86b9-b55838b42409.png">

After trying a lot, I found the user/contact combination that works for _this specific config_. User and contact definition relies on the instance configuration to properly work cht-core. 
Try this:
```
[
   {
    "password": "Medic1234",
    "username": "henrry1",
    "phone": "+2868917046",
    "place": "e0538e70-f1c8-5220-82a4-bd03665851dc",
    "type": "chw",
    "contact": {
        "first_name": "Herry",
        "last_name": "Yolk",
        "name": "Henrry Yolk",
        "sex": "male",
        "phone": "22 22 12 23",
        "role": "chw",
        "geolocalized": "true",
        "meta": {
            "created_by": "admin",
            "created_by_person_uuid": "",
            "created_by_place_uuid": ""
        },
        "type": "contact",
        "contact_type": "c62_chw"
     }
   },
   {
    "password": "Medic1234",
    "username": "patrik1",
    "phone": "+2868917046",
    "place": "e0538e70-f1c8-5220-82a4-bd03665851dc",
    "type": "chw",
    "contact": {
        "first_name": "Patrik",
        "last_name": "Sand",
        "name": "Patrik Sand",
        "sex": "male",
        "phone": "22 22 12 23",
        "role": "chw",
        "geolocalized": "true",
        "meta": {
            "created_by": "admin",
            "created_by_person_uuid": "",
            "created_by_place_uuid": ""
        },
        "type": "contact",
        "contact_type": "c62_chw"
     }
   }
 ]
```

**Comment by mrjones-plip** [comment]
@latin-panda - thanks for all the detailed notes! I do wonder with just checking out the branch and then running `node server.js` for API if it wasn't loading all the new code where as `grunt dev-api` would a "better job"? I'll try and report back.

I'll be sure to post back the API responses for all my testing! 

I think we should work on just the complete flow when testing this feature: CSV exported from this file imported via the new API.  To that end, I'll try port your JSON above to the [CSV template](https://docs.google.com/spreadsheets/d/1Et33wKpP8o9AtRe_LAVevYtN06ZuhF9ipod6m8F9Uho/edit#gid=872411844). Safe to assume that we use periods in column names to denote paths?  So the column title for `created_by_person_uuid` should be `contact.meta.created_by_person_uuid`, correct?

I also see in the JSON you don't specify the `place` for the contact only for the user.  Is this first `place` field shared between the two when they're created?

**Comment by latin-panda** [comment]
> JSON testing is helpful, but not applicable to our primary use case

True, however after parsing the csv, we're using the json, so if the json fails, the csv fails too. Same code :) 

> To that end, I'll try port your JSON above to the [CSV template](https://docs.google.com/spreadsheets/d/1Et33wKpP8o9AtRe_LAVevYtN06ZuhF9ipod6m8F9Uho/edit#gid=872411844) 

This is what it would look like: 
https://docs.google.com/spreadsheets/d/1HDjvqVStDnLO1QdQz_1UEocDEEQM1ylW-4wnhfhP9C8/edit?usp=sharing

> I also see in the JSON you don't specify the place for the contact only for the user. Is this first place field shared between the two when they're created?

Looking [at the docs](https://docs.communityhealthtoolkit.org/apps/reference/api/#post-apiv1users), I noticed that contact's didn't have it. And looking at the DB's doc, it has the hierarchy of places automatically added: 
<img width="577" alt="Screen Shot 2022-05-03 at 5 20 54 pm" src="https://user-images.githubusercontent.com/66472237/166471727-03e0bd52-e814-4eb9-b2ee-12b8365d61fd.png">

**Comment by mrjones-plip** [comment]
> [This is](
https://docs.google.com/spreadsheets/d/1HDjvqVStDnLO1QdQz_1UEocDEEQM1ylW-4wnhfhP9C8/edit?usp=sharing) what it would look like

Ah - that's great! Given we think CSV should be working, I'll try and get [the one template we're using with all the fields](https://docs.google.com/spreadsheets/d/1Et33wKpP8o9AtRe_LAVevYtN06ZuhF9ipod6m8F9Uho/edit#gid=872411844) to look like yours in hopes it works.

> Looking [at the docs](https://docs.communityhealthtoolkit.org/apps/reference/api/#post-apiv1users), I noticed that contact's didn't have it. And looking at the DB's doc, it has the hierarchy of places automatically added:

Ok cools - good news. As long as the contact is re-using the same UUID as is on the user, we should be all good!  I'll test this today as well.

**Comment by mrjones-plip** [comment]
OK!! I'm quite satisfied to have gotten an end to end go through of the [exact test steps](https://docs.google.com/document/d/1FKowuecQNwxOZrx1ktDgWRhl3yJnCMicNZHOt3ot9So/edit#heading=h.nimljpt9tx6o), following the happy path, and not getting lost!  The fixes I had to ensure were in place were:
1. use `--data-binary` with the `curl` call. Huge props to @tatilepizs  for discovering this. The call I ended up using, assuming your file is `7503.sample.3.csv`:
   ```
   curl --location --request POST 'http://medic:password@10.0.40.183:5988/api/v1/users' --header 'Content-Type: text/csv' --data-binary '@7503.sample.3.csv'
   ```
3. make sure column names are EXACT. I knew this, but still.  As well, make sure there's no errant chars on new rows. the VLOOKUP code i had was leaving a `Select Site ASC` and the API was naturally saying that every other field was missing (it was!). The [test template](https://docs.google.com/spreadsheets/d/1Et33wKpP8o9AtRe_LAVevYtN06ZuhF9ipod6m8F9Uho/edit) is up to date (and these 3 rows work!).
4. make sure my dev env was stable (like [a tortoise](https://medic.slack.com/archives/C0WDA7Y0J/p1651620744746939?thread_ts=1651616307.118019&cid=C0WDA7Y0J) ;) 

So now when follow the test plan exactly, as linked above, I get perfectly populated users and contacts associated to existing places.  It's really lovely! These rows are attaching to different places, so they're not bumping into [the issue](https://medic.slack.com/archives/C02GME3BQE6/p1651592374437719) that @latin-panda found earlier. 

Once we get a new build with the fix for setting primary contact (which we assume will be `/api/v2/users`, we should follow these same steps.

**Comment by mrjones-plip** [comment]
@latin-panda - just dropping a quick note to say I tested as of 851f2a2 and it's looking really good! I'm using `/api/v2/users` and I see the `medic-logs -> user-upload-EPOCH` documents getting created and updated as the import is going.  I did note that none of the contacts are being set as the default, but assume that is coming next :rocket:

**Comment by mrjones-plip** [comment]
Minor update: I suspect the audit log with complete record of all data imported is more work. If we're running out of time, we can skip this for the FR!

**Comment by latin-panda** [comment]
@mrjones-plip thanks for testing, the progress record in the logs is just 40% done which covers the parsing from csv to json only. What's still pending in the log work is: 
- log progress of saving users
- log data 
- make errors, because we have like 3 or 4 different structures, and log the errors to the respective user record so we know which one failed.  

> I did note that none of the contacts are being set as the default, but assume that is coming next 

I removed this piece on purpose, first it was giving trouble and second we can not default in an upload of 1000 users, I think the admin-user needs to specify which contact, if any, is the primary contact for X place, but this is a topic for latter.

**Comment by mrjones-plip** [comment]
@elvisdorkenoo  and @m5r - I see the tests are [failing in "Test It!"](https://github.com/medic/cht-core/runs/6332127598?check_suite_focus=true#step:14:636).  In case that log gets purged b/c of retention policies, here's the verbose log for Services started -> Users API -> token-login

   <details>
   <summary>CI log</summary>
   
```
1) Users API
    token-login
        when token-login is configured
        should create many users where one fails to be created w/o token_login:
    StatusCodeError: 404 - {"code":404,"error":"missing"}
    at new StatusCodeError (/home/runner/work/cht-core/cht-core/node_modules/request-promise-core/lib/errors.js:32:15)
    at /home/runner/work/cht-core/cht-core/node_modules/request-promise-core/lib/plumbing.js:97:41
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

2) Users API
    token-login
        when token-login is configured
        should fail to create many users with invalid fields w/o token_login:

    AssertionError: expected 404 to equal 400
    + expected - actual

    -404
    +400
    
    at Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/api/controllers/users.js:958:41)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

3) Users API
    token-login
        when token-login is configured
        should create and update many users correctly w/o token_login:
    StatusCodeError: 404 - {"code":404,"error":"missing"}
    at new StatusCodeError (/home/runner/work/cht-core/cht-core/node_modules/request-promise-core/lib/errors.js:32:15)
    at /home/runner/work/cht-core/cht-core/node_modules/request-promise-core/lib/plumbing.js:97:41
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

4) Users API
    token-login
        when token-login is configured
        should create and update many users correctly with token_login:
    StatusCodeError: 404 - {"code":404,"error":"missing"}
    at new StatusCodeError (/home/runner/work/cht-core/cht-core/node_modules/request-promise-core/lib/errors.js:32:15)
    at /home/runner/work/cht-core/cht-core/node_modules/request-promise-core/lib/plumbing.js:97:41
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```
   </details>

**Comment by m5r** [comment]
Merged to `master` with bulk user upload in #7678
